File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public static String solution(String[] cards1, String[] cards2, String[] goal) {
        StringBuilder sb = new StringBuilder();
        ArrayDeque<String> card1 = new ArrayDeque<>();
        ArrayDeque<String> card2 = new ArrayDeque<>();

        Collections.addAll(card1, cards1); // card1에 카드 다 넣기
        Collections.addAll(card2, cards2); // card2에 카드 다 넣기

        for (String s : goal) {
            if (!card1.isEmpty() && card1.peekFirst().equals(s)) {
                sb.append(card1.pollFirst());
                continue;
            }
            if (!card2.isEmpty() && card2.peekFirst().equals(s)) {
                sb.append(card2.pollFirst());
                continue;
            }
            return "No";
        }
        return "Yes";
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer = ''
    cards1.reverse()
    cards2.reverse()
    goal.reverse()
    
    while True:
        if not goal :
            return "Yes"
        if cards1 and cards1[-1] == goal[-1] :
            cards1.pop()
            goal.pop()
            continue
        if cards2 and cards2[-1] == goal[-1] :
            cards2.pop()
            goal.pop()
            continue
        return "No"